### ISSUE TITLE
[C++][Java]When i use DatasetFileWriter::write method write a file, I can't specify a file name

### ISSUE THREAD

**Issue by shouriken** [apache/arrow#43489]
### Describe the usage question you have. Please include as many useful details as  possible.

I use this static method to write a parquet file into fs, I give empty partition array, so it will be in a file; and i give the `baseNameTemplate` arg is "test.parquet" to specify the filename, but it leads to an error: `basename_template did not contain '{i}'`.
```java
public static void write(BufferAllocator allocator, ArrowReader reader, FileFormat format, String uri,
                           String[] partitionColumns, int maxPartitions, String baseNameTemplate) {
    try (final ArrowArrayStream stream = ArrowArrayStream.allocateNew(allocator)) {
      Data.exportArrayStream(allocator, reader, stream);
      JniWrapper.get().writeFromScannerToFile(stream.memoryAddress(),
          format.id(), uri, partitionColumns, maxPartitions, baseNameTemplate);
    }
  }
```
read the cpp jni code and cpp dataset's code, the FileSystemDatasetWriteOptions::basename_template seems not supported to specify the file name without `{i}`.
```cpp
/// \brief Options for writing a dataset.
struct ARROW_DS_EXPORT FileSystemDatasetWriteOptions {
.
.
.
  /// Template string used to generate fragment basenames.
  /// {i} will be replaced by an auto incremented integer.
  std::string basename_template;
.
.
.
}
```
If I have any method to specify the filename when none partition?

### Component(s)

Java

**Comment by vibhatha** [comment]
cc @felipecrv

**Comment by felipecrv** [comment]
This is the case because the names of files in a **dataset** have to follow a convention to ensure they can be read as a **dataset**. Maybe you should use directories and files directly to have your own conventions?

**Comment by shouriken** [comment]
Yes, I want to specify the path of target-file directly, write the whole dataset without column-partition to the specified target file. But I don't find a method to do it, is there any method ready for it?

**Comment by felipecrv** [comment]
@vibhatha knows more about the Java APIs. It looks like filesystem APIs aren't exposed yet in Arrow Java. (?)
